### PR TITLE
Update build tools and CLI version:

### PR DESCRIPTION
**Pull request by MattGal**
- Update some paths to Net46 for BuildTools update
 - Workaround lack of updated Microsoft.Private.PackageBaseline.  Created https://github.com/dotnet/corefx/issues/17675 to track this; once that's addressed we can roll this back and simply update the dependency in pkg\baseline\project.json
 - Add suppressions for reved inbox assms

@StephenBonikowsky : This passes build and packaging steps, please let me know what else might be useful to test.

### CODE REVIEW

File: src/System.ServiceModel.Security/pkg/System.ServiceModel.Security.pkgproj [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
+  <PropertyGroup>
+    <PreventImplementationReference>true</PreventImplementationReference>
```
Comment (mconnew) [inline src/System.ServiceModel.Security/pkg/System.ServiceModel.Security.pkgproj]: Why did you add this? My understanding is this prevents applications from referencing the package and is meant for usage with private packages.

**Comment by StephenBonikowsky** [comment]
Tests appear to be working on Windows but not on other OSes, looks like a common error...

`error MSB4019: The imported project "/mnt/resource/j/workspace/dotnet_wcf/master/centos7.1_debug_prtest/Tools/Microsoft.Net.Compilers/2.0.0-rc/tools/Microsoft.CSharp.Core.targets" was not found.`

File: src/System.ServiceModel.Security/pkg/System.ServiceModel.Security.pkgproj (hunk 1 above)
Comment (MattGal) [inline src/System.ServiceModel.Security/pkg/System.ServiceModel.Security.pkgproj]: Meant to remove that, was trying various stuff to get the packages fixed before I talked to Eric St. John.  Will remove.

File: src/dir.targets [hunk 2]
```diff
@@ -23,5 +23,8 @@
   <Target Name="DumpTargets" BeforeTargets="ResolveProjectReferences">
     <Message Text="DumpTargets> $(OutputPath), C=[$(Configuration)], CG=[$(ConfigurationGroup)], OG=[$(OSGroup)], TG=[$(TargetGroup)]" Importance="Low" />
   </Target>
+  
+  <!-- Workaround for issue seen after build tools update, some test projects mysteriously lose this target despite not needing it --> 
+  <Target Name="GetTargetFrameworkProperties"/>
```
Comment (MattGal) [inline src/dir.targets]: I spent a lot of time trying to track this down, since really nothing should work if the .targets that define this were missing (Tools/Microsoft.Common.CurrentVersion.targets) but this works for build, test, and packaging.  We should definitely figure it out at some point.

**Comment by MattGal** [comment]
@StephenBonikowsky depending on WCF's tolerance for my weird 1-line workaround, this looks ready to go.  I can keep investigating of not, it's just tricky as the investigation has to happen on *Nix where I don't have quite the same toolkit.

**Comment by StephenBonikowsky** [comment]
With all the tests now running and passing I'm ok with proceeding.
I'll setup a private run on VSTS.

**Comment by StephenBonikowsky** [comment]
All VSO runs look good now.

File: src/dir.targets (hunk 2 above)
Comment (weshaggard) [inline src/dir.targets]: Can you please file an issue for this?

File: pkg/baseline/packageIndex.json [hunk 3]
```diff
@@ -1,14 +1,2219 @@
 {
   "Packages": {
```
Comment (weshaggard) [inline pkg/baseline/packageIndex.json]: Shouldn't this be shared from corefx?

Comment (MattGal) [inline pkg/baseline/packageIndex.json]: It should, but around the time when the dev/eng branch merged in, we stopped pushing baseline packages to the feed.  @ericstj was looking into bringing this back.

File: src/dir.targets (hunk 2 above)
Comment (MattGal) [inline src/dir.targets]: https://github.com/dotnet/buildtools/issues/1417
